### PR TITLE
[12.0][MIG] hr_holidays_hour

### PR DESCRIPTION
**Pull request by astirpe**
Porting of `hr_holidays_hour` to V12.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Isn't Odoo now computing holidays in hours?

**Comment by astirpe** [comment]
In Odoo it is possible to configure the leave types to express the duration in hours or days. However in the leave summary report, in the employee form and in some other places, the duration is displayed in days regardless of the configuration of the leave type. This is confusing for the customers that are used to manage leaves in hours. I will improve the description of the module in order to clarify this.

**Comment by pedrobaeza** [comment]
OK then

**Comment by kittiu** [comment]
@astirpe thank you for your contribution. I am about to test this but found the following error when install this module.
```
Element '<field name="number_of_days">' cannot be located in parent view

Error context:
View `hr.leave tree`
[view_id: 417, xml_id: n/a, model: hr.leave, parent_id: 391]
None" while parsing /home/kittiu/workspace/temp/hr_holidays_hour/views/hr_leave.xml:4, near
<record id="hr_leave_view_tree" model="ir.ui.view">
        <field name="model">hr.leave</field>
        <field name="inherit_id" ref="hr_holidays.hr_leave_view_tree"/>
        <field name="arch" type="xml">
            <data><field name="number_of_days" position="attributes">
                <attribute name="invisible">1</attribute>
            </field>
            <field name="number_of_days" position="after">
                <field name="number_of_hours_display" sum="Total leaves"/>
            </field>
        </data></field>
    </record>
```
Note: I just  git pull odoo12 now, before do the install, I checked `number_of_days` is removed from `hr_holidays.hr_leave_view_tree`

**Comment by astirpe** [comment]
@kittiu thank you!

**Comment by astirpe** [comment]
One test of `hr_holidays_accrual_advanced` is failing. The error is not related to this PR because in the travis logs, module `hr_holidays_hour` is installed after `hr_holidays_accrual_advanced`.

**Comment by kittiu** [comment]
Functional testing. I am not very sure about this, but from common sense, I would expect that the detail should get me the same hours remaining (376 != 400).

**Button says 376**
image

**Summary says 400**
image

Also, now we we change the meaning of smart button to Hrs, should you add Hrs in the button label?

@astirpe beside the above. All the code is LGTM :+1:

**Comment by astirpe** [comment]
@kittiu 

The details and the button do not display the same hours remaining (376 != 400) because, in the Employee's Button, the leaves requests and allocations taken into account:

- must have the leave type with Mode different than "No allocation";
- the leaves requests and allocations must be approved by the manager.

In the Employee's Button that you mentioned in the screenshot, only the following allocation are taken into account:

```
160.00	Approved	Allocation Request	Overtime Compensation
160.00	Approved	Allocation Request	Legal Leaves for Peter Parker
 56.00	Approved	Allocation Request	International Tour
```

The following leave is not taken into account:

```
-16.00	05/20/2019 02:00:00	05/22/2019 02:00:00	Approved	Doctor Appointment
```
and the reason is that its type is "Sick Leaves" that has Mode="No allocation".

Also the following leave request is not taken into account:

```
-16.00	05/01/2019 02:00:00	05/03/2019 02:00:00	To Approve	Trip with Family
```
because is not yet approved.

**Comment by kittiu** [comment]
@astirpe OK, that sound.
For this is non-blocking, IMHO would be nice if button has label like "Leave Left (Hrs)".
> Also, now we we change the meaning of smart button to Hrs, should you add Hrs in the button label?

**Comment by astirpe** [comment]
@kittiu to me it's not necessary to add Hrs in the button label. In the readme it's specified that the Leaves Left smart button will display the duration in hours.

File: hr_holidays_hour/models/hr_employee.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+from odoo import api, fields, models
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    @api.multi
+    def _compute_leaves_count(self):
+        leaves = self.env['hr.leave'].read_group([
```
Comment (nikul-serpentcs) [inline hr_holidays_hour/models/hr_employee.py:12]: @astirpe  here use `search_count` method, So easy  counting and fast return result

same as for allocations

**Review by nikul-serpentcs — COMMENTED** [review]
Minor change as per my opinion

File: hr_holidays_hour/models/hr_employee.py (hunk 1 above)
Comment (pedrobaeza) [reply hr_holidays_hour/models/hr_employee.py:12]: And even better to use `read_group` for batch record compute.

Comment (astirpe) [reply hr_holidays_hour/models/hr_employee.py:12]: @nikul-serpentcs I don't understand why should I use `search_count`. In method `_compute_leaves_count()` I'm not counting the number of records, I'm counting the hours of leaves and allocations.

@pedrobaeza I also don't understand what are you proposing.

Comment (nikul-serpentcs) [reply hr_holidays_hour/models/hr_employee.py:12]: > @nikul-serpentcs I don't understand why should I use `search_count`. In method `_compute_leaves_count()` I'm not counting the number of records, I'm counting the hours of leaves and allocations.


Ok, Or You can search a record which is your Domain and sum of `number_of_hours`.
It's Not Blocking anywhere but improvements.

**Comment by astirpe** [comment]
@kittiu @nikul-serpentcs is this PR approved for you?

**Review by nikul-serpentcs — APPROVED** [review]
Only Code review LGTM :+1:

**Comment by WhiteTiger08** [comment]
What's missing in this PR to be merged?

**Comment by alexey-pelykh** [comment]
@WhiteTiger08 it needs a functional review

**Review by kittiu — APPROVED** [review]

**Comment by astirpe** [comment]
Can this one be merged?

**Pull request merged**